server/routes/friends/friend.js

```diff
@@ -0,0 +1,326 @@
+      user.friends.unshift(friendId);
+      //remove the friend from list of received requests
+      user.receivedFriendRequests.splice(userIndex, 1);
+
+      friend.friends.unshift(userId);
+      //remove user from friend's list of sent requests
+      friend.outgoingFriendRequests.splice(friendIndex);
+    } else if (user.outgoingFriendRequests.includes(friendId)) {
+      //If you've already made a request, you must wait till the friend accepts
+      //Cannot add them yourself
+      return res
+        .status(400)
+        .json({msg: "Please wait for the user to accept your request."});
+    } else {
+      //Cannot add a friend unless a request is made and approved
+      return res
+        .status(400)
+        .json({msg: "Cannot add friend. Make a friend request first"});
+    }
+    await user.save();
+    await friend.save();
+    res.json({msg: "Added friend"});
+  } catch (error) {
+    console.log(error.message);
+    if (error.kind == "ObjectId") {
+      //If the friend ID passed in does not exist in the database
+      return res.status(400).json({msg: "User does not exist."});
+    }
+    res.status(500).send("Server error");
+  }
+});
+
+//@route            DELETE /api/remove-friend/:id
+//@desc             Remove this friend or delete the sent friend request
+//@access           Private
+router.delete("/friend/:id", [authentication], async function (req, res) {
+  //The ID passed in the params will be the ID of the user who you want to add as a friend
+  //We have the current user's id stored as part of the authentication middleware
+  const friendId = req.params.id;
+  const userId = req.user.id;
+  try {
+    const user = await User.findById(userId);
+    const friend = await User.findById(friendId);
+
+    if (!friend) {
+      //If the friend ID passed in does not exist in the database
+      return res.status(400).json({msg: "User does not exist"});
+    }
+
+    //Ensure user and friend are friends
+    //If so, delete each other from friends array
+    const friendIndex = user.friends.indexOf(friendId);
+    const userIndex = friend.friends.indexOf(userId);
+
+    if (friendIndex !== -1 && userIndex !== -1) {
+      user.friends.splice(friendIndex, 1);
+      friend.friends.splice(userIndex, 1);
+    } else {
+      return res.status(400).json({msg: "Not friend of this user."});
+    }
+
+    await user.save();
+    await friend.save();
+    res.json({msg: "Friend removed successfully."});
+  } catch (error) {
+    console.log(error.message);
+    if (error.kind == "ObjectId") {
+      //If the friend ID passed in does not exist in the database
+      return res.status(400).json({msg: "User does not exist."});
+    }
+    res.status(500).send("Server error");
+  }
+});
+
+//@route            POST /api/friend-request/:id
+//@desc             Make this user a friend of the current user
+//@access           Private
+router.post("/friend-request/:id", [authentication], async function (req, res) {
```

Comment:
```suggestion
router.post("/outgoing-requests", [authentication], async function (req, res) {
```
with the friend_id coming from the body